### PR TITLE
feat: add modal to explain why users cannot delete items [DET-6998]

### PR DESCRIPTION
**Pull request by julian-determined-ai**
## Description

feat: add modal to explain why users cannot delete items [DET-6998]

The "Delete" button is no longer disabled when deleting a model or deregistering a model version. Instead a modal is shown explaining that only an admin or the user who created the item is able to delete it. 

20FB050E-AD2E-44BE-BC4E-181BA4B8FDA4

Screen Shot 2022-04-25 at 2 23 50 PM

## Test Plan

1. Using an admin account or the account of a user that has created the model or version, attempt to delete a model or deregister version. Everything should work as expected.
2. Using a non admin account or account of a user that did not create a model or version. Attempt to deregister a model or version. The "Delete" or "Deregister" button should be enabled. After clicking the button, a modal should show explaining that "Only the item creator or an admin can delete items from the model registry".

## Commentary (optional)

The `showCannotDelete` function has been defined several times in this PR but would seeming be more appropriate defined in a single place such as in a `helpers` file. However, it does not seem as though any such directory is currently defined. I wondered if that was due in an effort to explicitly separate function/utility concerns. However, if we do think this function should be defined in one place only I will refactor. 

update: The refactor mentioned above has been implemented.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-6998]: https://determinedai.atlassian.net/browse/DET-6998?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by EmilyBonar — CHANGES_REQUESTED** [review]
The code itself is fine, but you're correct that redefining the modal that many times is undesirable. Take a look in the `hooks/useModal` directory, and try using that. I think `useModalExperimentDelete.tsx` is a pretty simple one you could base this off of. You might also consider changing the text to be more general so that this modal could be applied outside the Model Registry; something like `Only the item creator or an admin can delete this item.`

Also, this is actually ticket 6998, not 6698.

File: webui/react/src/pages/ModelVersionDetails/ModelVersionHeader.tsx [hunk 1]
```diff
@@ -117,9 +118,10 @@ const ModelVersionHeader: React.FC<Props> = (
       },
       {
         danger: true,
-        disabled: !isDeletable,
+        disabled: false,
```
Comment (EmilyBonar) [inline webui/react/src/pages/ModelVersionDetails/ModelVersionHeader.tsx:121]: nit: this line can be removed

**Review by EmilyBonar — APPROVED** [review]
LGTM

File: webui/react/src/pages/ModelVersionDetails/ModelVersionHeader.tsx (hunk 1 above)
Comment (EmilyBonar) [reply webui/react/src/pages/ModelVersionDetails/ModelVersionHeader.tsx:121]: to be clear, I'm referring to the `disabled` line

Comment (julian-determined-ai) [reply webui/react/src/pages/ModelVersionDetails/ModelVersionHeader.tsx]: agreed! This has been updated

**Pull request merged**